addons/stock/models/stock_quant.py

```diff
@@ -378,6 +472,58 @@ def _onchange_serial_number(self):
             if message:
                 return {'warning': {'title': _('Warning'), 'message': message}}
 
+    def _apply_inventory(self):
+        moves = OrderedSet()
+        for quant in self:
+            # Create and vaidate a move so that the quant matches its `inventory_quantity`.
+            if quant.inventory_diff_quantity > 0:
+                move_vals = quant._get_inventory_move_values(quant.inventory_diff_quantity, quant.product_id.with_company(
+                    quant.company_id).property_stock_inventory, quant.location_id)
+            else:
+                move_vals = quant._get_inventory_move_values(-quant.inventory_diff_quantity, quant.location_id, quant.product_id.with_company(
```

Comment:
Nice to have but maybe change description "Product Quantity Updated" in case diff = 0

Reply:
Done. Description when diff=0 is now "Product Quantity Confirmed".